components/engine_traits/src/engines.rs

```diff
@@ -31,3 +32,7 @@ impl<K: KvEngine, R: RaftEngine> Engines<K, R> {
         self.kv.sync()
     }
 }
+
+pub trait EnginesFactory<K, R> {
```

Comment:
What's the purpose?

Reply:
it's used to provide a unified way for creating engines. Per discussion offline, I can remove it for now. will revisit it if it's needed later.